src/ompl/base/objectives/src/PathLengthOptimizationObjective.cpp

```diff
@@ -62,10 +64,14 @@ ompl::base::Cost ompl::base::PathLengthOptimizationObjective::motionCostHeuristi
     return motionCost(s1, s2);
 }
 
-#if OMPL_HAVE_EIGEN3
 ompl::base::InformedSamplerPtr ompl::base::PathLengthOptimizationObjective::allocInformedStateSampler(const ProblemDefinitionPtr probDefn, unsigned int maxNumberCalls) const
 {
-    // Make the direct path-length informed sampler and return
+    // Make the direct path-length informed sampler and return. If OMPL was compiled with Eigen, a direct version is available, if not a rejection-based technique can be used
+#if OMPL_HAVE_EIGEN3
     return boost::make_shared<PathLengthDirectInfSampler>(probDefn, maxNumberCalls);
-}
+#else
+    OMPL_ERROR("%s: Direct sampling of the path-length objective requires Eigen, but OMPL was compiled without Eigen support. If possible, please install Eigen and recompile OMPL. If this is not possible, you can manually create an instantiation of RejectionInfSampler to approximate the behaviour of direct informed sampling. ", description_.c_str());
+    // Return a null pointer to avoid compiler warnings
+    return ompl::base::InformedSamplerPtr();
```

Comment:
Two questions:
- Is `OMPL_ERROR(...)` or `throw Exception(...)` preferred here?
- Instead of being a runtime erorr, is there a way to make this a compile-time error that only gets raised if (1) the function is used and (2) `!OMPL_HAVE_EIGEN3`? The only example I could find only was [here] (http://stackoverflow.com/questions/18813731/compile-time-check-if-a-function-is-used-unused-c), but it requires the use of a dummy template argument (which struck me as not worth the complexity?).